uncertainty_engine_types/message.py

```diff
@@ -1,13 +1,29 @@
 from datetime import datetime
-from typing import Dict, Sequence, Union
+from typing import Dict, Literal, Sequence, Union
 
-from pydantic import BaseModel
+from pydantic import BaseModel, model_validator
 
 
 Value = Union[str, float, int]
+StructuredOutput = Dict[str, Union[Value, Sequence[Value], Dict[str, Value]]]
 
 
 class Message(BaseModel):
-    role: str
-    content: Union[str, Dict[str, Union[Value, Sequence[Value]]]]
+    role: Literal["instruction", "user", "engine"]
+    content: Union[str, StructuredOutput]
     timestamp: datetime
+
+    @model_validator(mode="before")
```

Comment:
I think a `@classmethod` decorator is needed here.
```suggestion
    @model_validator(mode="before")
    @classmethod
```